### ISSUE TITLE
[k8s] Support Nvidia GFD Labels for GPU type detection

### ISSUE THREAD

**Issue by romilbhardwaj** [skypilot-org/skypilot#2460]
To detect GPU type on the cluster, we currently support GKE labels and `skypilot.co/accelerators` labels created by our GPU labelling script (`python -m sky.utils.kubernetes.gpu_labeler`).

It would be good to add a `GPULabelFormatter` for [Nvidia GPU Feature Discovery](https://github.com/NVIDIA/gpu-feature-discovery). To do so, we will need a list of label values generated by Nvidia GFD for SkyPilot supported GPUs (e.g., `nvidia.com/gpu.product: A100-SXM4-40GB`). 

Related issue: https://github.com/NVIDIA/k8s-device-plugin/issues/739

**Comment by asaiacai** [comment]
Based on the code the maintainer in the referenced in the related issue, I want to say that the output from `nvidia.com/gpu.product` is exactly the same as what you get from running `nvidia-smi --query-gpu=name --format=csv,noheader,nounits` as done [here](https://github.com/skypilot-org/skypilot/blob/9de6d1494e55e2f3a0545b50c69bc05fcc2d421e/sky/utils/kubernetes/k8s_gpu_labeler_setup.yaml#L81)? the only difference being [there's dashes instead of spaces](https://github.com/NVIDIA/gpu-feature-discovery/blob/152fa93619e973043d936f19bf20bb465c1ab289/internal/lm/resource.go#L163). On my machine I get the following outputs
```bash
$ kubectl describe node | grep nvidia.com/gpu.product
                    nvidia.com/gpu.product=NVIDIA-H100-80GB-HBM3

$ nvidia-smi --query-gpu=name --format=csv,noheader,nounits
NVIDIA H100 80GB HBM3
```
so it's probably enough to use [similar logic as the labeler job ](https://github.com/skypilot-org/skypilot/blob/9de6d1494e55e2f3a0545b50c69bc05fcc2d421e/sky/utils/kubernetes/k8s_gpu_labeler_setup.yaml#L66) to compare against the canonical gpu names we have? 

I'll can draft up a PR for this. it was getting hard to remember to rerun the labeler job anytime I added a node to my k3s cluster but this way it just works directly with feature discovery. Somewhat resolves #3432 if people are running with the NVIDIA gpu operator

**Comment by romilbhardwaj** [comment]
Good point, if the `nvidia.com/gpu.product` label indeed uses `nvidia-smi --query-gpu=name --format=csv,noheader,nounits` with `-` instead of ` `, we should be able to have a `GPULabelFormatter` that parses GFD labels into SkyPilot canonical names. Would love to see a PR. Just noting that we will need to test it extensively, since having a half-functioning label formatter is worse than having users run the labelling script for consistency.
